crates/wasi-preview1-component-adapter/src/lib.rs

```diff
@@ -2396,7 +2434,7 @@ impl State {
     }
 
     #[cold]
-    fn new() -> &'static RefCell<State> {
+    fn new() -> &'static UnsafeCell<State> {
```

Comment:
I've now removed the `with_mut` and this outer `UnsafeCell`.

I think the scariest part now is the `fn descriptors_mut(&self) -> &mut Descriptors`, because I couldn't get `RefCell`/`RefMut` to work now that `get_input_stream` needs to return a `&Descriptor` instead of a `u32` that it can just copy out. I'll poke around at this to see if I can figure out a better option.